lib/src/util/logging.cc

```diff
@@ -3,44 +3,146 @@
+#include <boost/log/sources/severity_logger.hpp>
+#include <boost/log/sources/record_ostream.hpp>
+#include <boost/log/utility/setup/common_attributes.hpp>
+#include <boost/log/attributes/scoped_attribute.hpp>
+
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/date_time/gregorian/gregorian.hpp>
+
+#include <boost/smart_ptr/make_shared_object.hpp>
+
 #include <map>
+#include <fstream>
+#include <cstdint>
 
 namespace PCPClient {
 namespace Util {
 
 namespace lth_log = leatherman::logging;
+namespace sinks = boost::log::sinks;
+namespace expr = boost::log::expressions;
+namespace attrs = boost::log::attributes;
+
+static bool access_logger_enabled { false };
+
+//
+// PCP Access Logging - sink backend
+//
+
+BOOST_LOG_ATTRIBUTE_KEYWORD(access_outcome, "AccessOutcome", std::string)
 
-static void setupLoggingImp(std::ostream &stream,
+class access_writer : public sinks::basic_sink_backend<sinks::synchronized_feeding>
+{
+  public:
+    explicit access_writer(std::shared_ptr<std::ostream> sink_stream_ptr);
+    void consume(boost::log::record_view const& rec);
+
+  private:
+    std::shared_ptr<std::ostream> _sink_stream_ptr;
+};
+
+access_writer::access_writer(std::shared_ptr<std::ostream> sink_stream_ptr)
+    : _sink_stream_ptr {std::move(sink_stream_ptr)}
+{
+}
+
+void access_writer::consume(boost::log::record_view const &rec)
+{
+    auto timestamp = boost::log::extract<boost::posix_time::ptime>("TimeStamp", rec);
+    auto access_outcome = boost::log::extract<std::string>("AccessOutcome", rec);
+    *_sink_stream_ptr << '[' << boost::gregorian::to_iso_extended_string(timestamp->date())
+                      << ' ' << boost::posix_time::to_simple_string(timestamp->time_of_day())
+                      << "] " << access_outcome << std::endl;
+}
+
+//
+// PCP Access Logging - logger
+//
+
+void logAccess(std::string const& message, int line_num)
+{
+    if (access_logger_enabled) {
+        boost::log::sources::severity_logger<lth_log::log_level> slg;
+        static attrs::constant <std::string> namespace_attr{
+                "puppetlabs.pcp_client.connector"};
+        slg.add_attribute("AccessOutcome",
+                          attrs::constant<std::string>(message));
+        slg.add_attribute("Namespace", namespace_attr);
+        slg.add_attribute("LineNum", attrs::constant<int>(line_num));
+
+        // TODO(ale): make leatherman.logging's color_writer::consume ignore
+        // empty messages (and don't stream 'message' below) or check the log
+        // level; in alternative, allow configuring the color_writer sink
+        // filter to avoid logging AccessOutcome messages (something like
+        // `lth_sink->set_filter(!expr::has_attr(access_outcome));`) - note
+        // that the on_message callback and the record log level are checked
+        // by the logger, not by color_writer::consume. Otherwise, as it's now,
+        // the access message will be logged if the configured log level > info.
```

Comment:
Nope, it doesn't matter. Because the backend consume method doesn't check the log level, nor invoke the on_message callback. leatherman.logging does that in the logger., .

But yes, I couldn't get it working with the basic logger; only with the severity one.